python/cugraph/structure/graph_implementation/simpleGraph.py

```diff
@@ -0,0 +1,838 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cugraph.structure import graph_primtypes_wrapper
+from cugraph.structure.symmetrize import symmetrize
+from cugraph.structure.number_map import NumberMap
+import cugraph.dask.common.mg_utils as mg_utils
+import cudf
+import dask_cudf
+import cugraph.comms.comms as Comms
+import pandas as pd
+import numpy as np
+from cugraph.dask.structure import replication
+
+
+class simpleGraphImpl:
+
+    class EdgeList:
+        def __init__(self, source, destination, edge_attr=None):
+            self.edgelist_df = cudf.DataFrame()
+            self.edgelist_df["src"] = source
+            self.edgelist_df["dst"] = destination
+            self.weights = False
+            if edge_attr is not None:
+                self.weights = True
+                if type(edge_attr) is dict:
+                    for k in edge_attr.keys():
+                        self.edgelist_df[k] = edge_attr[k]
+                else:
+                    self.edgelist_df["weights"] = edge_attr
+
+    class AdjList:
+        def __init__(self, offsets, indices, value=None):
+            self.offsets = offsets
+            self.indices = indices
+            self.weights = value  # Should be a dataframe for multiple weights
+
+    class transposedAdjList:
+        def __init__(self, offsets, indices, value=None):
+            simpleGraphImpl.AdjList.__init__(self, offsets, indices, value)
+
+    class Properties:
+        def __init__(self, properties):
+            self.multi_edge = getattr(properties, 'multi_edge', False)
+            self.directed = properties.directed
+            self.renumbered = False
+            self.self_loop = None
+            self.isolated_vertices = None
+            self.node_count = None
+            self.edge_count = None
+
+    def __init__(self, properties):
+        # Structure
+        self.edgelist = None
+        self.adjlist = None
+        self.transposedadjlist = None
+        self.renumber_map = None
+        self.properties = simpleGraphImpl.Properties(properties)
+        self._nodes = {}
+
+        # TODO: Move to new batch class
+        # MG - Batch
+        self.batch_enabled = False
+        self.batch_edgelists = None
+        self.batch_adjlists = None
+        self.batch_transposed_adjlists = None
+
+    # Functions
+    def __from_edgelist(
```

Comment:
This method is very large and does a lot, and I wonder if we need to break out the individual steps (verify, renumber, symmetrize, etc.) into individual methods, especially when expert mode is supported. This would also allow us to test the individual steps in isolation easier.

Reply:
Can be addressed in upcoming PRs as symmetrization and renumbering is updated

Reply:
added fixme